orbit/pkg/platform/platform_windows.go

```
@@ -176,3 +178,124 @@ func GetProcessByName(name string) (*gopsutil_process.Process, error) {
+// list of supported OSes on the go-smbios documentation.
+// The windows go-smbios implementations calls to GetSystemFirmwareTable()
+func hardwareGetSMBiosUUID() (string, error) {
+	// SMBIOS data in operating system-specific location
+	streamReader, smBIOSRawData, err := smbios.Stream()
+	if err != nil {
+		return "", fmt.Errorf("failed to open stream: %v", err)
+	}
+
+	// Ensure that stream will be closed
+	defer streamReader.Close()
+
+	// Decode SMBIOS structures from the stream.
+	decoder := smbios.NewDecoder(streamReader)
+	structSMBIOSdata, err := decoder.Decode()
+	if err != nil {
+		return "", fmt.Errorf("failed to decode BIOS structures: %v", err)
+	}
+
+	// Determine SMBIOS version and table location from entry point
+	biosMajor, biosMinor, _ := smBIOSRawData.Version()
+
+	// SMBIOS constants from spec here - https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_3.1.1.pdf
+	const systemInformationType uint8 = 0x01 // System Information indicator
+	const minBiosStructSize uint8 = 0x1b     // Section 7.2 on the SMBIOS specification (0x1a min header size + null character)
+	const uuidOffset uint8 = 0x4             // UUID offset in System Information (Type 1) structure
+	const revMajorVersion int = 0x3          // SMBIOS revision that most of the current BIOS have - v3 specs were released in 2015
+	const minLegacyMajorVersion int = 0x2    // Minimum SMBIOS Major rev that supports UUID little-endian encoding
+	const minLegacyMinorVersion int = 0x6    // Minimum SMBIOS Minor rev that supports UUID little-endian encoding
+	const uuidSize int = 0x10                // UUID size is calculated with field offset value (0xA) + node field length (6 bytes)
+
+	var smBiosUUID string = ""
+
+	// Walking the obtained SMBIOS data
+	for _, rawBiosStruct := range structSMBIOSdata {
+		if (rawBiosStruct.Header.Type == systemInformationType) && (rawBiosStruct.Header.Length >= minBiosStructSize) {
```

Should we return an error if this loop never finds the matching data? eg. if `len(structSMBIOSdata) == 0` or none of the entries match the header type and length requirement?

---

good point; thanks for bringing it up. I've just fixed this in the code.